php/post-types/class-block-post.php

```diff
@@ -1117,8 +1117,9 @@ public function list_table_content( $column, $post_id ) {
 			$block      = new Block( $post_id );
 			$categories = get_block_categories( get_post() );
```

Comment:
Should we also include a check to see if the block has a category?
`if ( ! empty( $block->category ) )`

Reply:
Yes it's working fine if we add above conditional logic. But isset is already used in some conditional logic so i have added isset instead of not empty condition.

`if ( isset( $this->controls[ $control_name ] ) ) {`
`if ( isset( $icons[ $block->icon ] ) ) {`

Reply:
The problem is occurring because `$block->category` is equal to an empty string, and `$categories['']` is not set.

Checking if `$categories['']` is set does help, so we should absolutely keep that, but I'm just suggesting that it would also nice to check right after we instantiate the `Block` whether the category is empty. If it is, we can skip running `get_block_categories`, `wp_list_pluck`, etc.

Reply:
Like this: 

```php
$block = new Block( $post_id );

if ( ! empty( $block->category ) ) {
    $categories = get_block_categories( get_post() );
    $categories = wp_list_pluck( $categories, 'title', 'slug' );

    if ( isset( $categories[ $block->category ] ) ) {
        echo esc_html( $categories[ $block->category ] );
    }
}
```